page-factory-core/src/main/java/ru/sbtqa/tag/pagefactory/aspects/CriticalStepCheckAspect.java

```diff
@@ -21,106 +21,107 @@
 import ru.sbtqa.tag.pagefactory.optional.PickleStepCustom;
 import ru.sbtqa.tag.qautils.errors.AutotestError;
 
-import java.lang.reflect.Field;
-import java.util.Arrays;
-import java.util.Optional;
-
 @Aspect
 public class CriticalStepCheckAspect {
     private static final String STEP_FIELD_NAME = "step";
+    private static final String DEFINITION_FIELD_NAME = "definitionMatch";
+
     private static final String NON_CRITICAL_CATEGORY_NAME = "Non-critical failures";
     private static final String NON_CRITICAL_CATEGORY_MESSAGE = "Some non-critical steps are failed";
+
     private final Category nonCriticalCategory = new Category(NON_CRITICAL_CATEGORY_NAME,
             NON_CRITICAL_CATEGORY_MESSAGE, null,
             Arrays.asList(Status.PASSED.value()));
 
-    private ThreadLocal<Pair<PickleStep, Throwable>> currentFailedNonCritical = new ThreadLocal<>();
-
     @Pointcut("execution(* cucumber.runtime.StepDefinitionMatch.runStep(..))")
     public void runStep() {
     }
 
     @Pointcut("execution(* cucumber.runner.EventBus.send(..)) && args(event,..) && if()")
-    public static boolean sendStepFinished(Event event) {
-        return event instanceof TestStepFinished;
+    public static boolean sendStepFinished(TestStepFinished event) {
+        return !event.testStep.isHook();
```

Comment:
Does it really handle instanceOf validation on the fly?

Reply:
absolutely!